File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/Delphi/ConditionalFormatting/ConditionalFormattingDemoMain.pas
unit ConditionalFormattingDemoMain;

{$I cxVer.inc}

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms, Menus, Dialogs, cxGraphics, cxControls,
  cxLookAndFeels, cxLookAndFeelPainters, dxSkinsCore, dxSpreadSheetCore, dxSpreadSheetFunctions, dxSpreadSheetGraphics,
  dxSpreadSheetClasses, dxSpreadSheetTypes, dxSpreadSheet, BaseForm, dxCore, dxCoreClasses, dxHashUtils, ComCtrls,
  dxSpreadSheetCoreHistory, dxSpreadSheetPrinting, dxSpreadSheetFormulas, cxClasses, dxSpreadSheetConditionalFormatting,
  dxSpreadSheetConditionalFormattingRules, dxSpreadSheetCoreStyles, dxSpreadSheetCoreStrs, dxSpreadSheetContainers,
  dxSpreadSheetHyperlinks, dxSpreadSheetStyles, dxSpreadSheetUtils, cxSplitter, dxSpreadSheetFormulaBar;

type
  { TfrmConditionalFormatting }

  TfrmConditionalFormatting = class(TfmBaseForm)
    SpreadSheet: TdxSpreadSheet;
    miConditionalFormatting: TMenuItem;
    miManageRules: TMenuItem;
    FormulaBar: TdxSpreadSheetFormulaBar;
    FormulaBarSplitter: TcxSplitter;

    procedure FormCreate(Sender: TObject);
    procedure miManageRulesClick(Sender: TObject);
  public
    function GetSpreadSheet: TdxSpreadSheet; override;
  end;

var
  frmConditionalFormatting: TfrmConditionalFormatting;

implementation

{$R *.dfm}

{$IFDEF EXPRESSTREELIST}
uses
  dxSpreadSheetConditionalFormattingRulesManagerDialog;
{$ENDIF}

const
  sdxWarningText = 'Conditional formatting dialogs are only available if the ExpressQuantumTreeList Suite is installed.';

{ TfrmConditionalFormatting }

procedure TfrmConditionalFormatting.FormCreate(Sender: TObject);
begin
  inherited;
  SpreadSheet.LoadFromFile('..\..\Data\TopTradingPartners.xlsx');
  WindowState := wsMaximized;
end;

function TfrmConditionalFormatting.GetSpreadSheet: TdxSpreadSheet;
begin
  Result := SpreadSheet;
end;

procedure TfrmConditionalFormatting.miManageRulesClick(Sender: TObject);
begin
{$IFDEF EXPRESSTREELIST}
  ShowConditionalFormattingRulesManagerDialog(SpreadSheet);
{$ELSE}
  MessageDlg(sdxWarningText, mtWarning, [mbOK], 0);
{$ENDIF}
end;

end.

File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/CBuilder/ConditionalFormatting/ConditionalFormattingDemoMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "ConditionalFormattingDemoMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BaseForm"
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma link "dxCore"
#pragma link "dxCoreClasses"
#pragma link "dxHashUtils"
#pragma link "dxSpreadSheet"
#pragma link "dxSpreadSheetClasses"
#pragma link "dxSpreadSheetCore"
#pragma link "dxSpreadSheetFormulas"
#pragma link "dxSpreadSheetFunctions"
#pragma link "dxSpreadSheetGraphics"
#pragma link "dxSpreadSheetTypes"
#pragma link "dxSpreadSheetCoreHistory"
#pragma link "dxSpreadSheetPrinting"
#pragma link "dxSpreadSheetConditionalFormatting"
#pragma link "dxSpreadSheetConditionalFormattingRules"
#pragma link "dxSpreadSheetConditionalFormattingRulesManagerDialog"
#pragma link "dxSpreadSheetFormulaBar"
#pragma link "cxSplitter"
#pragma resource "*.dfm"
TfrmConditionalFormatting *frmConditionalFormatting;
//---------------------------------------------------------------------------
__fastcall TfrmConditionalFormatting::TfrmConditionalFormatting(TComponent* Owner)
	: TfmBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmConditionalFormatting::FormCreate(TObject *Sender)
{
  TfmBaseForm::FormCreate(Sender);
  SpreadSheet->LoadFromFile("..\\..\\Data\\TopTradingPartners.xlsx");
  WindowState = wsMaximized;
}
//---------------------------------------------------------------------------

TdxSpreadSheet* TfrmConditionalFormatting::GetSpreadSheet()
{
  return SpreadSheet;
}
//---------------------------------------------------------------------------
void __fastcall TfrmConditionalFormatting::miManageRulesClick(TObject *Sender)
{
	ShowConditionalFormattingRulesManagerDialog(SpreadSheet);
}


